go/src/koding/kites/kloud/metrics/metrics.go

```diff
@@ -93,3 +88,55 @@ func (p *Publisher) Publish(r *kite.Request) (interface{}, error) {
 func (p *Publisher) Close() error {
 	return p.conn.Close()
 }
+
+func publishToCountly(url string, r []byte, log kite.Logger) error {
+	req := bytes.NewReader(r)
+
+	ticker := backoff.NewTicker(newBackoff())
+	defer ticker.Stop()
+
+	var err error
+	var retry bool
+	for range ticker.C {
+		// each request should not take more than 5 sec
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+
+		retry, err = shouldRetry(ctxhttp.Post(ctx, nil, url, "application/json", req))
```

Comment:
Attempt to read from req would fail with io.EOF on retry, you'd need `req = bytes.NewReader(r)` at L111.